packages/components/src/navigation/stories/index.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Navigation from '..';
+import NavigationGroup from '../group';
+import NavigationItem from '../item';
+import NavigationLevel from '../level';
+
+export default {
+	title: 'Components/Navigation',
+	component: Navigation,
+};
+
+const Container = styled.div`
+	max-width: 246px;
+`;
+
+function Example() {
+	const [ activeItem, setActiveItem ] = useState( 'item-1' );
+	const [ activeLevel, setActiveLevel ] = useState( 'root' );
+
+	return (
+		<Container>
+			<Navigation
+				activeItem={ activeItem }
+				activeLevel={ activeLevel }
+				setActiveItem={ setActiveItem }
+				setActiveLevel={ setActiveLevel }
+			>
+				<NavigationLevel title="Home">
+					<NavigationGroup title="Group 1">
+						<NavigationItem item="item-1" title="Item 1" />
+						<NavigationItem item="item-2" title="Item 2" />
+						<NavigationItem
+							badge="2"
+							item="item-3"
+							navigateToLevel="category"
+							title="Category"
+						/>
+					</NavigationGroup>
+					<NavigationGroup title="Group 2">
+						<NavigationItem
+							href="https://wordpress.org/"
+							item="item-4"
+							target="_blank"
+							title="External link"
+						/>
+						<NavigationItem item="item-5">
+							<img
+								alt="WordPress Logo"
+								src="https://s.w.org/style/images/about/WordPress-logotype-wmark-white.png"
+								style={ { width: 50, height: 50 } }
+							/>
+						</NavigationItem>
+					</NavigationGroup>
+				</NavigationLevel>
+
+				<NavigationLevel
+					level="category"
+					parentLevel="root"
+					parentLevelTitle="Home"
+					title="Category"
+				>
+					<ul>
+						<NavigationItem
+							badge="1"
+							item="child-1"
+							title="Child 1"
+						/>
+						<NavigationItem item="child-2" title="Child 2" />
+						<NavigationItem
+							navigateToLevel="nested-category"
+							item="child-3"
+							title="Nested Category"
+						/>
+					</ul>
+				</NavigationLevel>
+
+				<NavigationLevel
+					level="nested-category"
+					parentLevel="category"
+					parentLevelTitle="Category"
+					title="Nested Category"
+				>
+					<ul>
+						<NavigationItem
+							item="sub-child-1"
+							title="Sub Child 1"
+						/>
+						<NavigationItem
+							item="sub-child-2"
+							title="Sub Child 2"
+						/>
+					</ul>
+				</NavigationLevel>
+
+				<NavigationLevel title="Secondary Menu">
```

Comment:
What's the benefit of allowing two `NavigationLevel` in one view, and can't that be handled by existing `NavigationGroup`s instead?

Reply:
`Level` and `Group` define different hierarchies.

`Level` can be toggled when navigating through the menu hierarchy, so they are the nested _levels_ of the navigation.
`Group` are a way to group items of the same `Level` into titled sub-lists.

Having multiple `Level` on the same _level_ is a bit confusing, I'll give you that, but I don't see issues in _allowing_ it. 🤔 
I'll start by removing the multile `Level` in the story, so that we won't be actively recommending it to consumers.

Reply:
> Having multiple Level on the same level is a bit confusing, I'll give you that, but I don't see issues in allowing it.

Yeah, that's mostly what I was hinting at. I'd expect singular level within one view/screen, partly because of the `navigateToLevel` functionality. This way we are not navigating just to the level, but to something more generic that can currently contain multiple levels, in which case that name is not completely accurate.

Reply:
>Having multiple Level on the same level is a bit confusing, I'll give you that, but I don't see issues in allowing it.

In order to have a separation between groups without having to mark `height: 100%` on a bunch of divs, multiple levels comes in handy.

![Screen Shot 2020-09-11 at 1 29 12 PM](https://user-images.githubusercontent.com/1922453/92838507-fbe74000-f432-11ea-907b-d073f80ec4f0.png)